client/src/main/java/io/atomix/copycat/client/util/ClientConnection.java

```diff
@@ -203,14 +203,15 @@ public ClientConnection reset(Address leader, Collection<Address> servers) {
         return connectFuture;
       }
 
-      connectFuture = new OrderedCompletableFuture<>();
-      connectFuture.whenComplete((r, e) -> connectFuture = null);
+      CompletableFuture<Connection> future = new OrderedCompletableFuture<>();
+      connectFuture.whenComplete((r, e) -> this.connectFuture = null);
```

Comment:
Yeah, and there was actually a bug in `OrderedCompletableFuture` that was shown in tests. Tests should pass now.